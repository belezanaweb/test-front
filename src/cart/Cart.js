import React, { Component } from 'react';

class Cart extends Component {
  render() {
    return (
      <div>
        Cart
      </div>
    );
  }
}

export default Cart;