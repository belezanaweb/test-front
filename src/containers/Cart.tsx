import * as React from 'react';
import { Connection, Inject } from 'exredux';
import { modelStore } from '../service/modelStore';
import { ProductListModel } from '../service/models/ProductListModel';
import { Title } from '../components/Title';
import { CartProducts } from '../components/CartProducts';
import { CartSummary } from '../components/CartSummary';
import { Button } from '../components/Button';
import { Wrapper } from '../components/Wrapper';
import { Link } from 'react-router-dom';
// ---------------------------------------------------------------------
// --- COMPONENT -------------------------------------------------------
// ---------------------------------------------------------------------
class Props {
  @Inject productList?: ProductListModel;
}

@Connection({
  modelStore,
  props: Props
})
export class Cart extends React.Component<Props> {
  componentDidMount() {
    const { productList } = this.props;
    productList.getProducts();
  }

  render() {
    const { productList } = this.props;

    if (productList.isFailed) {
      return <div>Falha ao listar produtos!</div>;
    }

    if (productList.isLoading) {
      return <div>Carregando produtos...</div>;
    }

    if (productList.isCompleted) {
      const products = productList.response.data;
      return (
        <Wrapper>
          <Title>Produtos</Title>
          <CartProducts items={products.items} />
          <CartSummary
            discount={products.discount}
            shippingTotal={products.shippingTotal}
            subTotal={products.subTotal}
            total={products.total}
          />
          <Link to="payment">
            <Button>Seguir para o pagamento</Button>
          </Link>
        </Wrapper>
      );
    }

    return null;
  }
}
