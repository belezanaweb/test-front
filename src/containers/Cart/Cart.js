import React, { useContext, useEffect, Suspense } from 'react'
import { useAsyncResource } from 'use-async-resource'
import { Link } from 'react-router-dom'
import { CartContext } from '../../store/CartStore'
import { Button } from '../../components/Button/Button'
import { ProductList } from '../../components/ProductList/ProductList'
import { PurchaseSummary } from '../../components/PurchaseSummary/PurchaseSummary'
import { Stepper } from '../../components/Stepper/Stepper'
import { Layout, LayoutGrid, LayoutAside } from '../../components/Layout/Layout'

const fetchCart = () =>
  fetch('http://www.mocky.io/v2/5b15c4923100004a006f3c07').then((res) => res.json())

const Cart = () => {
  const cartContext = useContext(CartContext)
  const [cartReader] = useAsyncResource(fetchCart, '')

  const data = cartReader()

  useEffect(() => {
    if (!cartContext.state.items) {
      cartContext.dispatch(data)
    }
  }, [cartContext, data])

  return (
    <Layout>
      <Stepper steps={steps} />
      <LayoutGrid>
        <Suspense fallback={<div>Carregando...</div>}>
          <ProductList />
          <LayoutAside>
            <PurchaseSummary />
            <Link to="/pagamento">
              <Button>Seguir para o pagamento</Button>
            </Link>
          </LayoutAside>
        </Suspense>
      </LayoutGrid>
    </Layout>
  )
}

const steps = [
  {
    active: true,
    title: 'Sacola'
  },
  {
    title: 'Pagamento'
  },
  {
    title: 'Confirmação'
  }
]

export default Cart
