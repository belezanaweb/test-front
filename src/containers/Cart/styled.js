import styled from 'styled-components'

import * as T from 'constants/styles/typography'
import * as C from 'constants/styles/colors'

export const CartContainer = styled.div`
  /* ... */
`
