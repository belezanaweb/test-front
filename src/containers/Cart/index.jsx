/* eslint-env browser */
import React, { useEffect } from 'react';
import { connect, useDispatch, useSelector } from 'react-redux';
import { Container, Row, Col } from 'react-grid-system';
import { Link } from 'react-router-dom';
import { getCartData } from 'store/cart/actions';
import DefaultLayout from 'layouts/default';

import Text from 'components/Text';
import Button from 'components/Button';
import Product from 'components/Product';
import Card from 'components/Card';
import Checkout from 'components/Checkout';

import ContainerStyled from '../style';

const Cart = () => {
  useEffect(() => {
    document.title = `Beleza na Web - Sacola`;
  }, []);
  const dispatch = useDispatch();
  const data = useSelector(state => state.cart.data);

  useEffect(() => {
    if (!data) {
      dispatch(getCartData());
    }
  }, [dispatch, data]);

  return (
    <ContainerStyled className="animated fadeIn faster">
      <Container>
        {data.items && data.items.length > 0 ? (
          <Row>
            <Col lg={8} xs={12}>
              <Text
                size="14px"
                className="product-text"
                uppercase
                bold
                color="#999"
              >
                Produtos
              </Text>
              <Card>
                {data.items.map(item => (
                  <Product showPrice key={item.product.sku} item={item} />
                ))}
              </Card>
            </Col>
            <Checkout data={data} />
            <Col xs={12}>
              <Link
                to="/payment"
                className="animated fadeIn"
                data-cy="buttonCart"
              >
                <Button>Seguir para o pagamento</Button>
              </Link>
            </Col>
          </Row>
        ) : (
          <Text>Sem items no carrinho.</Text>
        )}
      </Container>
    </ContainerStyled>
  );
};

export default DefaultLayout(connect()(Cart));
