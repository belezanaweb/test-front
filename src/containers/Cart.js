/** @jsx jsx */
import { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { jsx } from '@emotion/core'
import styled from '@emotion/styled'
import * as actionCart from '../actions/cart'

import Card from '../components/cart/card'

const Container = styled('div')`
  height: 720px;
  width: 360px;
  background-color: #EEE;
`
const CardContainer = styled('div')`
  height: 325px;
  width: 341px;
  border-radius: 3px;
  background-color: #FFF;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
`;


class Cart extends Component {

  state = {}

  componentDidMount() {
    this.handleFetchDate()
  }
  handleFetchDate = () => {
    const { fetchProductsCart } = this.props

    fetchProductsCart()
  }

  render() {
    const { cart } = this.props
    return (
      <Container>
        <CardContainer>
          {
            cart.items.map((item, index) =>
            <Card items={item} key={index} />
        )}
        </CardContainer>
      </Container>
    );
  }
}

Cart.propTypes = {
  cart: PropTypes.object,
  fetchProductsCart: PropTypes.func.isRequired
}

const mapStateToProps = ({ cartReducer }) => {
  return {
    cart: cartReducer.cart
  }
}

const mapDispatchToProps = dispatch => {
	return bindActionCreators(actionCart, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
