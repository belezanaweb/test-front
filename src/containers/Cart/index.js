import React, { useEffect } from 'react';
import { connect } from "react-redux";
import CardProduct from '../../components/CardProduct';
import styled from 'styled-components';
import Button from '../../components/Button';
import CardTotalShipping from '../../components/CardTotalShipping';
import { listCart } from '../../actions/cart';
import { routes } from "../Router";
import { push } from "connected-react-router";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #EEE;
  height: 720px;
`

const CartPage = props => {

  useEffect(() => {
    props.listCart()
  })

  return (
    <Container>
      <CardProduct products={props.cart.items}/>
      <CardTotalShipping />
      <Button name="SEGUIR PARA O PAGAMENTO" onClick={props.goToPayment}/>
    </Container> 
  )
}

const mapStateToProps = state => {
  return {
    cart: state.cart.cart
  }
}

const mapDispatchToProps = dispatch => ({
  listCart: () => dispatch(listCart()),
  goToPayment: () => dispatch(push(routes.payment))
})

export default connect(mapStateToProps, mapDispatchToProps)(CartPage)