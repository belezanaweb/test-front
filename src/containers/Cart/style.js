import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;

  @media (min-width: 1024px) {
    display: flex;
    justify-content: center;
    flex-direction: row;
    align-items: flex-start;
    height: 100%;
  }
`;

export default Container;
