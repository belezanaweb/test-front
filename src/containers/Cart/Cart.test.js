import React, { Suspense } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, screen, waitFor } from '@testing-library/react'
import { CartStore } from '../../store/CartStore'
import Cart from './Cart'

const CartStoreValue = {
  items: [
    {
      product: {
        name: 'Teste',
        priceSpecification: {
          price: 123.45
        },
        imageObjects: [
          {
            small: 'http://localhost.test'
          }
        ]
      }
    }
  ],
  subTotal: 123.45,
  total: 432.1,
  shippingTotal: 345.67,
  discount: 37.02
}

test('renders Cart', async () => {
  fetch.mockResponseOnce(JSON.stringify(CartStoreValue))

  const { container } = render(
    <Router>
      <Suspense fallback={<div>Testing...</div>}>
        <CartStore>
          <Cart />
        </CartStore>
      </Suspense>
    </Router>
  )

  expect(container).toMatchSnapshot()

  await waitFor(() => screen.getByText('Seguir para o pagamento'))

  expect(container).toMatchSnapshot()
})
