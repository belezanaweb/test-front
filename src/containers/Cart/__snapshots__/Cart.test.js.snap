// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cart 1`] = `
<div>
  <div>
    Testing...
  </div>
</div>
`;

exports[`renders Cart 2`] = `
<div>
  .c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 18px;
  width: 100%;
  min-height: 60px;
  border: 0;
  border-radius: 3px;
  background-color: #ff6c00;
  color: #fff;
  font-size: 20px;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.2;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 rgba(0,0,0,0.25);
  cursor: pointer;
}

.c20:hover,
.c20:active {
  background-color: #d45a00;
  box-shadow: none;
}

.c20:disabled {
  background-color: #ccc;
  box-shadow: none;
}

.c7 {
  padding: 12px 12px 13px 13px;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
}

.c6 {
  display: block;
  margin: 0 0 5px 13px;
  color: #999;
  font-size: 14px;
  font-weight: 700;
  line-height: 1.21428571428571;
  text-transform: uppercase;
}

.c8 {
  display: grid;
  grid-gap: 15px;
}

.c9 {
  display: grid;
  grid-gap: 11px 15px;
  grid-template-columns: 65px 1fr;
  grid-template-rows: 1fr 1fr;
  padding: 12px 10px 14px;
  border: 1px solid #eee;
  border-radius: 3px;
}

.c10 {
  width: 65px;
  height: 65px;
  object-fit: contain;
  object-position: 50% 100%;
  grid-row: span 2;
}

.c11 {
  margin: 0;
  color: #212122;
  font-size: 13px;
  font-weight: normal;
  line-height: 1.23076923076923;
}

.c12 {
  display: block;
  font-size: 14px;
  font-weight: 700;
  line-height: 1.21428571428571;
  text-align: right;
}

.c14 {
  display: grid;
  grid-gap: 9px;
  padding: 15px 13px 13px 14px;
  border: 1px solid #ccc;
  border-radius: 3px;
  color: #212122;
  font-size: 14px;
  line-height: 1.21428571428571;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 0;
}

.c16 {
  text-transform: uppercase;
}

.c17 {
  text-align: right;
  margin: 0;
}

.c18 {
  color: #ff7800;
}

.c19 {
  margin-top: 7px;
  font-weight: 700;
}

.c1 {
  width: 100%;
  margin-bottom: 10px;
  padding: 12px 0;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1024px;
  margin: 0 auto;
  padding: 0 20px;
  list-style: none;
}

.c3 {
  display: block;
  color: #ccc;
  font-size: 13px;
  font-weight: 700;
  line-height: 1.23076923076923;
  text-transform: uppercase;
  color: #ff7800;
}

.c4 {
  display: block;
  color: #ccc;
  font-size: 13px;
  font-weight: 700;
  line-height: 1.23076923076923;
  text-transform: uppercase;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: grid;
  grid-gap: 20px;
  padding: 0 9px 20px;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
}

.c13 {
  display: grid;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
  grid-gap: 20px;
}

@media only screen and (min-width:768px) {
  .c5 {
    margin-top: 40px;
    grid-template-columns: 1fr 340px;
  }
}

@media only screen and (min-width:768px) {
  .c13 {
    margin-top: 20px;
  }
}

<main
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
        >
          Sacola
        </li>
        <li
          class="c4"
        >
          Pagamento
        </li>
        <li
          class="c4"
        >
          Confirmação
        </li>
      </ul>
    </div>
    <div
      class="c5"
    >
      <section
        class=""
      >
        <h1
          class="c6"
        >
          Produtos
        </h1>
        <div
          class="c7 c8"
        >
          <article
            class="c9"
          >
            <img
              alt="Teste"
              class="c10"
              src="http://localhost.test"
            />
            <h2
              class="c11"
            >
              Teste
            </h2>
            <span
              class="c12"
            >
              R$ 123.45
            </span>
          </article>
        </div>
      </section>
      <aside
        class="c13"
      >
        <section
          class="c14"
        >
          <dl
            class="c15"
          >
            <dt
              class="c16"
            >
              Produtos
            </dt>
            <dd
              class="c17"
            >
              <span>
                R$ 123.45
              </span>
            </dd>
          </dl>
          <dl
            class="c15"
          >
            <dt
              class="c16"
            >
              Frete
            </dt>
            <dd
              class="c17"
            >
              <span>
                R$ 345.67
              </span>
            </dd>
          </dl>
          <dl
            class="c15 c18"
          >
            <dt
              class="c16"
            >
              Desconto
            </dt>
            <dd
              class="c17"
            >
              − 
              <span>
                R$ 37.02
              </span>
            </dd>
          </dl>
          <dl
            class="c15 c19"
          >
            <dt
              class="c16"
            >
              Total
            </dt>
            <dd
              class="c17"
            >
              <span>
                R$ 432.10
              </span>
            </dd>
          </dl>
        </section>
        <a
          href="/pagamento"
        >
          <button
            class="c20"
          >
            Seguir para o pagamento
          </button>
        </a>
      </aside>
    </div>
  </main>
</div>
`;
